packages/api/src/platforms/vtex/utils/skuVariants.ts

```diff
@@ -0,0 +1,168 @@
+import { StoreProduct as StoreProductType } from '../../..'
+import type { Product, Item } from '../clients/search/types/ProductSearchResult'
+
+export type SkuVariants = StoreProductType[]
+
+export type SkuVariantsByName = Record<
+  string,
+  Array<{ alt: string; src: string; label: string; value: string }>
+>
+
+function findSkuVariantImage(availableImages: Item['images']) {
+  return (
+    availableImages.find(
+      (imageProperties) => imageProperties.imageLabel === 'skuvariation'
+    ) ?? availableImages[0]
+  )
+}
+
+export function createSlugsMap(
+  variants: Item[],
+  mainVariant: string,
+  baseSlug: string
+) {
+  /**
+   * Maps property value combinations to their respective SKU's slug. Enables
+   * us to retrieve the slug for the SKU that matches the currently selected
+   * variations in O(1) time.
+   *
+   * Example: `'Color-Red-Size-40': 'classic-shoes-37'`
+   */
+  const slugsMap: Record<string, string> = {}
+
+  variants.forEach((variant) => {
+    const skuSpecificationProperties = variant.variations
+
+    if (skuSpecificationProperties.length === 0) {
+      return
+    }
+
+    // Make sure that the 'name-value' pair for the `mainVariant` variation
+    // is always the first one.
+    let skuVariantKey = `${mainVariant}-${
+      skuSpecificationProperties.find(
+        (variationDetails) => variationDetails.name === mainVariant
+      )?.values[0] ?? ''
+    }`
+
+    skuSpecificationProperties.forEach((property) => {
+      skuVariantKey +=
+        property.name !== mainVariant
+          ? `-${property.name}-${property.values[0]}`
+          : ''
+    })
+
+    slugsMap[skuVariantKey] = `${baseSlug}-${variant.itemId}`
+  })
+
+  return slugsMap
+}
+
+export function getActiveSkuVariations(variations: Item['variations']) {
+  const activeVariations: Record<string, string> = {}
+
+  variations.forEach((variation) => {
+    activeVariations[variation.name] = variation.values[0]
+  })
+
+  return activeVariations
```

Comment:
Kinda nitpicking, the `.reduce()` version could be:
```suggestion
return variations.reduce((activeVariations, variation) => {
    activeVariations[variation.name] = variation.values[0]
    return activeVariations
  }, activeVariations)
```

Reply:
This is the same case as the outermost from the previous comment. So let's bypass it 😉